Agents/EmailAgent/email-agent-code/src/main/java/uk/ac/cam/cares/jps/agent/email/EmailAgent.java

```diff
@@ -0,0 +1,291 @@
+package uk.ac.cam.cares.jps.agent.email;
+
+import static uk.ac.cam.cares.jps.agent.email.EmailAgentConfiguration.KEY_WHITE_IPS;
+import static uk.ac.cam.cares.jps.agent.email.EmailAgentConfiguration.KEY_WHITE_ONLY;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+import javax.servlet.ServletException;
+import javax.servlet.UnavailableException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.BadRequestException;
+import org.json.JSONObject;
+import org.springframework.stereotype.Controller;
+import uk.ac.cam.cares.jps.base.agent.JPSAgent;
+
+/**
+ * This EmailAgent uses the JPS Asynchronous Watcher framework and listens for incoming HTTP
+ * requests. Once a valid HTTP request has been received (and the originating source is approved),
+ * then the contents are forward onto an SMTP server (of the developer's choosing) to be dispatched
+ * as an email.
+ *
+ * A class (EmailSender) has been provided within the JPS Base Library to facilitate this. If the
+ * EmailAgent cannot be reached (or the request is not approved), then the contents are written to a
+ * local log file instead.
+ *
+ * @author Michael Hillman
+ */
+@Controller
+@WebServlet(urlPatterns = {"/email"})
+public class EmailAgent extends JPSAgent {
+
+    /**
+     * Is the EmailAgent in a valid state.
+     */
+    private boolean validState = true;
+
+    /**
+     * Perform required setup.
+     *
+     * @throws ServletException if in invalid state
+     */
+    @Override
+    public void init() throws ServletException {
+        super.init();
+
+        // Read the properties file
+        try {
+            EmailAgentConfiguration.readProperties();
+        } catch (IOException ioException) {
+            System.out.println("ERROR: Could not read properties!");
+            ioException.printStackTrace(System.out);
+            validState = false;
+
+            throw new UnavailableException("EmailAgent is not in valid state, could not read properties.");
+        }
+    }
+
+    @Override
+    public JSONObject processRequestParameters(JSONObject requestParams) {
+        return new JSONObject();
+    }
+
+    /**
+     * Processes HTTP requests.
+     *
+     * @param requestParams Request parameters in a JSONObject
+     * @param request HTTP Servlet Request
+     *
+     * @return result JSON
+     */
+    @Override
+    public JSONObject processRequestParameters(JSONObject requestParams, HttpServletRequest request) {
+        System.out.println("processRequestParameters(2)");
```

Comment:
`Logger` used in the `EmailHandler` class, can we use it instead of `System.out` here too?

Reply:
Yes, I must have missed this when I changed from Loggers to System.out.

Reply:
Logger removed.